cpp/include/cudf_test/testing_main.hpp

```diff
@@ -181,14 +200,12 @@ inline auto make_stream_mode_adaptor(cxxopts::ParseResult const& cmd_opts)
  * function parses the command line to customize test behavior, like the
  * allocation mode used for creating the default memory resource.
  */
-#define CUDF_TEST_PROGRAM_MAIN()                                        \
-  int main(int argc, char** argv)                                       \
-  {                                                                     \
-    ::testing::InitGoogleTest(&argc, argv);                             \
-    auto const cmd_opts = parse_cudf_test_opts(argc, argv);             \
-    auto const rmm_mode = cmd_opts["rmm_mode"].as<std::string>();       \
-    auto resource       = cudf::test::create_memory_resource(rmm_mode); \
-    rmm::mr::set_current_device_resource(resource.get());               \
-    auto adaptor = make_stream_mode_adaptor(cmd_opts);                  \
-    return RUN_ALL_TESTS();                                             \
+#define CUDF_TEST_PROGRAM_MAIN()                                  \
+  int main(int argc, char** argv)                                 \
+  {                                                               \
+    ::testing::InitGoogleTest(&argc, argv);                       \
+    auto const cmd_opts = parse_cudf_test_opts(argc, argv);       \
+    auto mr             = make_memory_resource_adaptor(cmd_opts); \
+    auto adaptor        = make_stream_mode_adaptor(cmd_opts);     \
```

Comment:
IMO these variables should be marked as `[[maybe_unused]]` to prevent warning-as-errors that may be issued by any future versions of GCC.